File: app/src/main/kotlin/jb/openware/app/components/SearchBarView.java
package in.afi.codekosh.components;

import android.app.Activity;
import android.content.Context;
import android.text.TextWatcher;
import android.view.MenuItem;
import android.view.View;
import android.view.inputmethod.EditorInfo;
import android.view.inputmethod.InputMethodManager;

import androidx.appcompat.widget.Toolbar;

import com.google.android.material.appbar.MaterialToolbar;

import in.afi.codekosh.R;

public class SearchBarView {
    private final Activity activity;
    private MaterialToolbar toolbar;
    private SearchEditText editText;

    public SearchBarView(Activity activity) {
        this.activity = activity;
    }

    public void init(MaterialToolbar toolbar, SearchEditText editText) {
        this.editText = editText;
        this.toolbar = toolbar;
    }

    public void setMenuVisibility(boolean isVisible) {
        if (!isVisible) {
            toolbar.getMenu().clear();
        } else {
            toolbar.inflateMenu(R.menu.profile_menu);
        }
    }


    public void setToolbarClickListener(View.OnClickListener listener) {
        toolbar.setNavigationOnClickListener(listener);
    }

    public void editTextOpener() {
        editText.requestFocus();
        InputMethodManager imm = (InputMethodManager) activity.getSystemService(Context.INPUT_METHOD_SERVICE);
        if (imm != null) {
            imm.showSoftInput(editText, InputMethodManager.SHOW_IMPLICIT);
        }
    }

    public void setHint(String hint) {
        editText.setHint(hint);
    }

    public void setOnTextChangedListener(TextWatcher textWatcher) {
        if (editText != null) {
            editText.addTextChangedListener(textWatcher);
        }
    }

    public void setEditTextEnable(boolean enable) {
        if (!enable) {
            editText.setFocusable(false);
            editText.setClickable(true);
            editText.setCursorVisible(false);
        } else {
            editText.setFocusable(true);
            editText.setClickable(true);
        }
    }

    public void setEditTextClickListener(View.OnClickListener listener) {
        if (editText != null) {
            editText.setOnClickListener(listener);
        }
    }

    public void setSearchSubmitListener(View.OnClickListener listener) {
        if (editText != null) {
            editText.setOnEditorActionListener((v, actionId, event) -> {
                if (actionId == EditorInfo.IME_ACTION_SEARCH) {
                    listener.onClick(v);
                    return true;
                }
                return false;
            });
        }
    }

    public String getSearchText() {
        return String.valueOf(editText.getText());
    }

    public void setMenuClickListener(Toolbar.OnMenuItemClickListener listener) {
        toolbar.setOnMenuItemClickListener(listener);
    }

    public MenuItem getMenuIcon() {
        return toolbar.getMenu().findItem(R.id.profile);
    }


}

File: app/src/main/kotlin/jb/openware/app/ui/components/SearchBarView.kt
package jb.openware.app.ui.components

import android.app.Activity
import android.text.TextWatcher
import android.view.MenuItem
import android.view.View
import android.view.inputmethod.EditorInfo
import android.view.inputmethod.InputMethodManager
import androidx.appcompat.widget.Toolbar
import androidx.core.content.getSystemService
import com.google.android.material.appbar.MaterialToolbar
import jb.openware.app.R

class SearchBarView(
    private val activity: Activity,
    private val toolbar: MaterialToolbar,
    private val editText: SearchEditText
) {

    fun setMenuVisibility(isVisible: Boolean) {
        toolbar.menu.clear()
        if (isVisible) {
            toolbar.inflateMenu(R.menu.profile_menu)
        }
    }

    fun setToolbarClickListener(listener: View.OnClickListener) {
        toolbar.setNavigationOnClickListener(listener)
    }

    fun openEditText() {
        editText.requestFocus()
        val imm: InputMethodManager? = activity.getSystemService()
        imm?.showSoftInput(editText, InputMethodManager.SHOW_IMPLICIT)
    }

    fun setHint(hint: CharSequence) {
        editText.hint = hint
    }

    fun setOnTextChangedListener(textWatcher: TextWatcher) {
        editText.addTextChangedListener(textWatcher)
    }

    fun setEditTextEnabled(enabled: Boolean) {
        editText.apply {
            isFocusable = enabled
            isClickable = true
            isCursorVisible = enabled
            if (enabled) {
                isFocusableInTouchMode = true
            }
        }
    }

    fun setEditTextClickListener(listener: View.OnClickListener) {
        editText.setOnClickListener(listener)
    }

    fun setSearchSubmitListener(listener: View.OnClickListener) {
        editText.setOnEditorActionListener { v, actionId, _ ->
            if (actionId == EditorInfo.IME_ACTION_SEARCH) {
                listener.onClick(v)
                true
            } else {
                false
            }
        }
    }

    val searchText: String
        get() = editText.text?.toString().orEmpty()

    fun setMenuClickListener(listener: Toolbar.OnMenuItemClickListener) {
        toolbar.setOnMenuItemClickListener(listener)
    }

    val menuIcon: MenuItem?
        get() = toolbar.menu.findItem(R.id.profile)
}